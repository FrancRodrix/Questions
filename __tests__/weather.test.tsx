import "react-native";
import React from "react";
import Weather from "../src/Screens/weather";
import renderer from "react-test-renderer";

const navigation = {
  navigation: jest.fn(),
  params: {
    forecastData: {
      current: {
        temperature: "Delhi",
        humidity: "India",
        cloudcover: "12345",
        weather_descriptions: ["Sunny"],
      },
      location: {
        name: "delhi",
      },
    },
  },
};
const tree = renderer.create(<Weather route={navigation} />);
test("renders correctly", () => {
  expect(tree).toMatchSnapshot();
});
