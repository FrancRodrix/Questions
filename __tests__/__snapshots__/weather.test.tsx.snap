// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "left": 50,
        "opacity": 1,
        "position": "absolute",
        "top": 70,
      }
    }
  >
    <Text>
      GO BACK
    </Text>
  </View>
  <View
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "#3D85C6",
        "borderRadius": 10,
        "borderWidth": 1,
        "padding": 10,
        "width": "80%",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 18,
          "fontWeight": "600",
          "paddingBottom": 10,
          "textAlign": "center",
        }
      }
    >
      CLIMATE CONDITIONS
    </Text>
    <Text
      style={
        Object {
          "fontSize": 14,
          "paddingVertical": 3,
          "textAlign": "center",
        }
      }
    >
       TEMPERATURE:
      Delhi
    </Text>
    <Text
      style={
        Object {
          "fontSize": 14,
          "paddingVertical": 3,
          "textAlign": "center",
        }
      }
    >
       HUMIDITY:
      India
    </Text>
    <Text
      style={
        Object {
          "fontSize": 14,
          "paddingVertical": 3,
          "textAlign": "center",
        }
      }
    >
      CLOUDCOVER
      12345
    </Text>
    <Text
      style={
        Object {
          "fontSize": 14,
          "paddingVertical": 3,
          "textAlign": "center",
        }
      }
    >
      WEATHER DISCRIPTION:
      Sunny
    </Text>
    <Text
      style={
        Object {
          "fontSize": 14,
          "paddingVertical": 3,
          "textAlign": "center",
        }
      }
    >
      CITY:
      delhi
    </Text>
  </View>
</RCTSafeAreaView>
`;
